### PR TITLE
feat: legislator badge (#1157)

### PR DESCRIPTION
**Pull request by 1emu**
Closes #1157 

- Legislator Badge
- AidropJob Entity
- Job for running AirdropJobs

### CODE REVIEW

File: src/components/AddressSelect/AddressesSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+})
+
+interface Props {
+  maxAddressesAmount: number
+  allowLoggedUserAccount?: boolean
+  setUsersAddresses: (addresses?: string[]) => void
+  isDisabled?: boolean
+}
+
+function AddressesSelect({ setUsersAddresses, isDisabled, maxAddressesAmount, allowLoggedUserAccount = false }: Props) {
```
Comment (andyesp) [inline src/components/AddressSelect/AddressesSelect.tsx:44]: I think we can create a common component where logic from this component and CoAuthorSelect can be reused.

File: src/components/Debug/BadgesAdmin.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  return (
+    <div className={className}>
+      <ContentSection>
+        <Heading size="sm">{'Badges'}</Heading>
+        <div>
+          <Button className="Debug__SectionButton" primary disabled={formDisabled} onClick={() => handleAirdropBadge()}>
+            {'Airdrop'}
+          </Button>
+          <Button className="Debug__SideButton" primary disabled={formDisabled} onClick={() => handleRevokeBadge()}>
```
Comment (andyesp) [inline src/components/Debug/BadgesAdmin.tsx:69]: We can hide this one if we don't plan to support it now.

File: src/entities/Coauthor/model.ts [hunk 3]
```diff
@@ -41,4 +43,19 @@ export default class CoauthorModel extends Model<CoauthorAttributes> {
     `
     await this.query(query)
   }
+
+  static async findAllCoauthors(proposals: ProposalAttributes[], status?: CoauthorStatus): Promise<string[]> {
```
Comment (andyesp) [inline src/entities/Coauthor/model.ts:47]: Maybe `findAllFromProposals`?

File: src/services/BadgesService.ts [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
+import { airdrop } from '../entities/Badges/utils'
+import CoauthorModel from '../entities/Coauthor/model'
+import { CoauthorStatus } from '../entities/Coauthor/types'
+import { ProposalAttributes, ProposalType } from '../entities/Proposal/types'
+import { getChecksumAddress } from '../entities/Snapshot/utils'
 import { ErrorCategory } from '../utils/errorCategories'
 
 import { ErrorService } from './ErrorService'
 
+const TRANSACTION_UNDERPRICED = -32000
```
Comment (andyesp) [inline src/services/BadgesService.ts:18]: `TRANSACTION_UNDERPRICED_ERROR_CODE`?

File: src/services/BadgesService.ts [hunk 5]
```diff
@@ -52,4 +64,74 @@ export class BadgesService {
...
+      return false
+    }
+  }
+
+  static async giveLegislatorBadges(acceptedProposals: ProposalAttributes[]) {
+    const governanceProposals = acceptedProposals.filter((proposal) => proposal.type === ProposalType.Governance)
+    const coauthors = await CoauthorModel.findAllCoauthors(governanceProposals, CoauthorStatus.APPROVED)
+    const authors = governanceProposals.map((proposal) => proposal.user)
+    const authorsAndCoauthors = new Set([...authors.map(getChecksumAddress), ...coauthors.map(getChecksumAddress)])
+    await this.queueAirdropJob(LEGISLATOR_BADGE_SPEC_CID, Array.from(authorsAndCoauthors))
```
Comment (andyesp) [inline src/services/BadgesService.ts:136]: If this is going to a queue, do we need to await it?

Comment (1emu) [reply src/services/BadgesService.ts:136]: Yes, if we don't await and something goes wrong while saving the job to the DB, we won't know (it happened to me while testing)

File: src/components/AddressSelect/AddressesSelect.tsx (hunk 1 above)
Comment (1emu) [reply src/components/AddressSelect/AddressesSelect.tsx:44]: Yes we can and should. 
It started as a dev tool and now it's like a relevant feature, but I'd rather do this in a different PR, WDYT?

File: src/components/Debug/BadgesAdmin.tsx (hunk 2 above)
Comment (1emu) [reply src/components/Debug/BadgesAdmin.tsx:69]: I need to test this before releasing 😬

File: src/components/AddressSelect/AddressesSelect.tsx (hunk 1 above)
Comment (1emu) [reply src/components/AddressSelect/AddressesSelect.tsx:45]: [#1183](https://github.com/decentraland/governance/issues/1183)

**Review by andyesp — APPROVED** [review]

**Pull request merged**